lib/puppet/provider/service/systemd.rb

```diff
@@ -46,11 +67,20 @@ def status
   end
 
   def enable
+    output = systemctl("unmask", @resource[:name])
```

Comment:
Well, that was a weird dilemma.

My logic was this:

1) Puppet is the authoritative source of truth
2) If Puppet says to enable it, then enable it
3) Setting something to state 'unmasked', but not started, while valid is
probably not what anyone wants.

I suppose that we could use the :mask parameter and you *must* set that to
:false to have it unmask. However, in theory, that's what the default would
be anyway.

So, in that vein, we could have enable respect the :mask parameter and
:mask would override enable.

Thoughts?

On Wed, Feb 11, 2015 at 1:26 PM, Kylo Ginsberg <notifications@github.com>
wrote:

> In lib/puppet/provider/service/systemd.rb
> <https://github.com/puppetlabs/puppet/pull/3141#discussion_r24519150>:
>
> > @@ -46,11 +67,20 @@ def status
> >    end
> >
> >    def enable
> > +    output = systemctl("unmask", @resource[:name])
>
> Might doing a 'unmask' here defeat someone's explicit intention that the
> service be left un-enable-able?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/3141/files#r24519150>.
>



-- 
Trevor Vaughan
Vice President, Onyx Point, Inc
(410) 541-6699
tvaughan@onyxpoint.com

-- This account not approved for unencrypted proprietary information --